lib/ansible/plugins/terminal/nxos.py

```diff
@@ -48,6 +50,47 @@ class TerminalModule(TerminalBase):
         re.compile(br"invalid (.+?)at '\^' marker", re.I)
     ]
 
+    def on_become(self, passwd=None):
+        if self._get_prompt().endswith(b'enable#'):
+            return
+
+        out = self._exec_cli_command(b'show privilege')
+        out = to_text(out, errors='surrogate_then_replace').strip()
+        if 'Disabled' in out:
+            raise AnsibleConnectionFailure('Feature privilege is not enabled')
+
+        # if already at privilege level 15 return
+        if '15' in out:
+            return
+
+        cmd = {u'command': u'enable'}
+        if passwd:
+            cmd[u'prompt'] = to_text(r"(?i)[\r\n]?Password: $", errors='surrogate_or_strict')
+            cmd[u'answer'] = passwd
+            cmd[u'prompt_retry_check'] = True
```

Comment:
else part need to be handled with exception - password is mandatory or don't make passed argument optional in function definition

Reply:
We wouldn't want to do anything if passwd is None because the user is not going to use `become_pass` if password is disabled for enable mode for the device.